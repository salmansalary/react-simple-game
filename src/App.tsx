import React, { useEffect, useState } from "react";
import { BrowserRouter as Router, Route, Switch, Redirect } from "react-router-dom";
import { AppProvider } from "./hooks/appHook";
import "./index.scss";
import Home from "./routes/home";
import Play from "./routes/play";
import { audioObject } from "./utils";
/*
    The preLoadImages is helping on deployed and production Mode,
    Since i have some number of graphic assets and images i preload
    them to browser cache to provide smooth game journey
*/
async function preLoadImages() {
	new Image().src = await (await import("./assets/images/1.png")).default;
	new Image().src = await (await import("./assets/images/2.png")).default;
	new Image().src = await (await import("./assets/images/3.png")).default;
	new Image().src = await (await import("./assets/images/4.png")).default;
	new Image().src = await (await import("./assets/images/5.png")).default;
	new Image().src = await (await import("./assets/images/6.png")).default;
	new Image().src = await (await import("./assets/images/7.png")).default;
	new Image().src = await (await import("./assets/images/8.png")).default;
	new Image().src = await (await import("./assets/images/backImg.png")).default;
	new Image().src = await (await import("./assets/images/backImgMain.png")).default;
	new Image().src = await (await import("./assets/images/hole.png")).default;

	await audioObject.loadFiles();
}

function App() {
	const [loading, setLoading] = useState(true);

	useEffect(() => {
		//Redirect to home on hard refresh if user manually reload the page
		if (window.location.pathname === "/play") {
			window.location = window.location.origin as any;
			return;
		}

		preLoadImages().then(() => {
			setLoading(false);
		});
	}, []);

	return (
		<>
			{loading && (
				<div className="loaderContainer">
					<label className="loader">Loading ...</label>
				</div>
			)}
			<AppProvider>
				<Router>
					<Switch>
						<Route exact path="/home" component={Home} />
						<Route path="/play" component={Play} />
						<Redirect from="/" to="/home" />
					</Switch>
				</Router>
			</AppProvider>
		</>
	);
}

export default App;
