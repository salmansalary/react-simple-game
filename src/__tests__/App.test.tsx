import React from 'react';
import { shallow, mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import App from '../App';

describe('App', () => {

    it('should render App correctly', () => {
        const component = shallow(<App />);
    
        expect(component).toMatchSnapshot();
    });

    it('should navigate correctly after press start with valid NAME input', async () => {
       
        

        await act(async function() {

            const component = mount(<App />);
    
            //Wait for resource preload
            await new Promise(resolve => setTimeout(()=>resolve(true),1000))
            component.update()

            let input = component.find('input.nameInput');
            input.instance().value = "SALMAN"
            input.simulate('change');
    
            const startBtn = component.find('button.btnStart');
            startBtn.simulate('click');
    
            expect(global.window.location.pathname).toEqual('/play');

        });
        
    });

});

