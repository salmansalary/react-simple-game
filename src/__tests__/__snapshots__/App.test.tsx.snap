// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render App correctly 1`] = `ShallowWrapper {}`;
